### PR TITLE
Vulnerability Detector integration tests for Amazon Linux 2022

### LINKED ISSUE #2813: Integration tests for Vulnerability Detector: Amazon Linux 2022 support
### Description

This issue is part of wazuh/wazuh#13034

We will include support for the Amazon Linux 2022 OS, including a new feed – ALAS 2022.

The following tests should cover the new supported OS version:

- [Feed tests](https://github.com/wazuh/wazuh-qa/tree/master/tests/integration/test_vulnerability_detector/test_feeds): These tests have to cover the addition of the new feed for `Amazon Linux 2022`
- [Provider tests](https://github.com/wazuh/wazuh-qa/tree/master/tests/integration/test_vulnerability_detector/test_providers): Here we will test the settings for `Amazon Linux` with OS version `2022`
- [Scan results tests](https://github.com/wazuh/wazuh-qa/tree/master/tests/integration/test_vulnerability_detector/test_scan_results): Here we have to add the tests that ensure `Amazon Linux 2022` agents are scanned properly

### PR DESCRIPTION
**Pull request by Deblintrake09**
|Related issue|
|---|
|Close #2813  |

## Description

For Wazuh v4.4.0 we want to add Amazon Linux 2022 OS support for the Wazuh Vulnerability Detector module.

The following test was added to cover the new supported OS version:

- `Feed tests:` These tests have to cover the addition of the new feed for `Amazon Linux 2022`
- `Provider tests`: Here we test the settings for `Amazon Linux` with OS version `2022`
- `Scan results tests`: Here we  add the tests that ensure `Amazon Linux 2022` agents are scanned properly

Modules Involved: 

- test_vulnerability_detector/test_feeds/test_download_feeds.py
- test_vulnerability_detector/test_feeds/test_duplicate_feeds.py
- test_vulnerability_detector/test_feeds/test_import_invalid_feed_type.py
- test_vulnerability_detector/test_providers/test_enabled.py
- test_vulnerability_detector/test_providers/test_missing_os.py
- test_vulnerability_detector/test_providers/test_os.py
- test_vulnerability_detector/test_providers/test_update_interval.py
- test_vulnerability_detector/test_scan_results/test_scan_provider_and_nvd_vulnerabilities.py
- test_vulnerability_detector/test_scan_results/test_scan_provider_vulnerabilities.py
- test_vulnerability_detector/test_scan_results/test_scan_vulnerability_removal.py

## Tests Results
| Test Path | Os/Type | Execution Type | Results | Date | By |
|--|--|--|--|--|--|
|  test/integration/test_vulnerability_detector/| CentOS - Manager | Jenkins | [:green_circle: ](https://ci.wazuh.info/job/Test_integration/27725)  [:green_circle: ](https://ci.wazuh.info/job/Test_integration/27739)  [:green_circle: ](https://ci.wazuh.info/job/Test_integration/27740) | 2022/06/09 | @Deblintrake09  | 
|  test/integration/test_vulnerability_detector/| CentOS - Manager | Local | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8764512/R3.zip) [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8764510/R1.zip)  [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/8764511/R2.zip)  | 2022/05/24 | @Deblintrake09  | 

## Tests

- [x] Proven that tests **pass** when they have to pass.
- [x] Proven that tests **fail** when they have to fail.

- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] Python codebase is documented following the Google Style for Python docstrings.

### CODE REVIEW

File: tests/integration/test_vulnerability_detector/test_feeds/data/test_cases/cases_download_feeds.yaml [hunk 1]
```diff
@@ -156,6 +156,17 @@
...
 
+- name: 'ALAS-2022'
+  description: 'Amazon Linux provider'
+  configuration_parameters:
+    PROVIDER: 'alas'
+    OS: 'amazon-linux-2022'
+  metadata:
+    provider_name: 'Amazon Linux 2022'
+    provider_os: 'Amazon-Linux-2022'
+    download_timeout: 120
```
Comment (damarisg) [inline tests/integration/test_vulnerability_detector/test_feeds/data/test_cases/cases_download_feeds.yaml:167]: I don't remember if I asked this in the previous PR, but Did you verify that the `download_timeout` value is correct?

File: tests/system/provisioning/agentless_cluster/roles/master-role/tasks/main.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@
       - policycoreutils
       - automake
       - autoconf
-      - libtool
+      - libtool
```
Comment (damarisg) [inline tests/system/provisioning/agentless_cluster/roles/master-role/tasks/main.yml:18]: Could you explain why this appears here, please?

**Review by damarisg — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,14 +1,24 @@
...
+
+Wazuh version: 4.5.0 \
+Wazuh commit: TBD \
+Release report: TBD
+
+### Added
+
+### Changed
+
+
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply CHANGELOG.md]: Added accidentaly by leaving in an incorrect commit
Fixed in https://github.com/wazuh/wazuh-qa/pull/2955/commits/24682dbc3c43af67dec1c5362b72fee96257aa15

File: tests/system/provisioning/agentless_cluster/roles/master-role/tasks/main.yml (hunk 2 above)
Comment (Deblintrake09) [reply tests/system/provisioning/agentless_cluster/roles/master-role/tasks/main.yml]: Fixed in https://github.com/wazuh/wazuh-qa/pull/2955/commits/546dcf98325b3a94eb580acc997fa4cc44463938

**Review by damarisg — APPROVED** [review]

File: tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml [hunk 4]
```diff
@@ -8,6 +8,27 @@
     provider_name: 'Amazon Linux 1'
     update_interval: '5s'
 
+- name: 'Amazon Linux 2'
```
Comment (jmv74211) [inline tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml:11]: Is it really necessary to add these extra use cases? I think one per operating system for this test is fine.

File: tests/integration/test_vulnerability_detector/test_feeds/test_download_feeds.py [hunk 5]
```diff
@@ -31,6 +31,7 @@
     - Arch Linux
     - Amazon Linux 2
     - Amazon Linux 1
+    - Amazon Linux 2022
```
Comment (jmv74211) [inline tests/integration/test_vulnerability_detector/test_feeds/test_download_feeds.py:34]:
```suggestion
    - Amazon Linux 2022
    - Amazon Linux 2
    - Amazon Linux 1
```

File: tests/integration/test_vulnerability_detector/test_feeds/test_duplicate_feeds.py [hunk 6]
```diff
@@ -31,6 +31,7 @@
     - Arch Linux
     - Amazon Linux 2
     - Amazon Linux 1
+    - Amazon Linux 2022
```
Comment (jmv74211) [inline tests/integration/test_vulnerability_detector/test_feeds/test_duplicate_feeds.py:34]:
```suggestion
    - Amazon Linux 2022
    - Amazon Linux 2
    - Amazon Linux 1
```

File: tests/integration/test_vulnerability_detector/test_feeds/test_import_invalid_feed_type.py [hunk 7]
```diff
@@ -29,6 +29,7 @@
     - Arch Linux
     - Amazon Linux 2
     - Amazon Linux 1
+    - Amazon Linux 2022
```
Comment (jmv74211) [inline tests/integration/test_vulnerability_detector/test_feeds/test_import_invalid_feed_type.py:32]:
```suggestion
    - Amazon Linux 2022
    - Amazon Linux 2
    - Amazon Linux 1
```

**Review by jmv74211 — CHANGES_REQUESTED** [review]

File: tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml (hunk 4 above)
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml]: Fixed in commit https://github.com/wazuh/wazuh-qa/pull/2955/commits/3e908b9a5f383bac61a7c00b9b37e80fd6e7004a

File: tests/integration/test_vulnerability_detector/test_feeds/test_import_invalid_feed_type.py (hunk 7 above)
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_feeds/test_import_invalid_feed_type.py]: Fixed in https://github.com/wazuh/wazuh-qa/pull/2955/commits/47bba0cb4c062403163cf3acfe3ea5a9f2220ddb

File: tests/integration/test_vulnerability_detector/test_feeds/test_duplicate_feeds.py (hunk 6 above)
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_feeds/test_duplicate_feeds.py]: Fixed In

File: tests/integration/test_vulnerability_detector/test_feeds/test_download_feeds.py (hunk 5 above)
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_feeds/test_download_feeds.py]: Fixed in https://github.com/wazuh/wazuh-qa/pull/2955/commits/47bba0cb4c062403163cf3acfe3ea5a9f2220ddb

**Review by jmv74211 — APPROVED** [review]

**Pull request merged**